src/components/frame/v5/pages/StreamingPaymentsPage/partials/StreamingPaymentsTable/partials/StreamingPaymentFilters/partials/TokenFilters/TokenFilters.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { type FC } from 'react';
+
+import { useStreamingFiltersContext } from '~frame/v5/pages/StreamingPaymentsPage/partials/StreamingPaymentsTable/FiltersContext/StreamingFiltersContext.ts';
+import { formatText } from '~utils/intl.ts';
+import { multiLineTextEllipsis } from '~utils/strings.ts';
+import Checkbox from '~v5/common/Checkbox/index.ts';
+import { TokenAvatar } from '~v5/shared/TokenAvatar/TokenAvatar.tsx';
+
+import { useGetTokenTypeFilters } from './hooks.ts';
+
+const TokenFilters: FC = () => {
+  const { tokenTypes, handleTokenTypesFilterChange } =
+    useStreamingFiltersContext();
+  const tokenTypesFilters = useGetTokenTypeFilters();
+  const tokenItems = tokenTypesFilters.map(({ token }) => ({
+    symbol: token.symbol,
+    label: (
+      <div className="flex items-center gap-2">
+        <TokenAvatar
+          size={18}
+          tokenName={token.name}
+          tokenAddress={token.tokenAddress}
+          tokenAvatarSrc={token.avatar ?? undefined}
+        />
+        {multiLineTextEllipsis(token.symbol, 5)}
+      </div>
+    ),
+    name: token?.tokenAddress || '',
+  }));
+
+  return (
+    <div>
+      <h5 className="hidden px-3.5 pb-2 uppercase text-gray-400 text-4 sm:block">
+        {formatText({ id: 'balancePage.filter.approvedTokenTypes' })}
+      </h5>
+      <ul>
+        {tokenItems.map(({ label, name }) => {
```

Comment:
I don't think it's necessary to map over all tokens and then map over them again, let's just map over `tokenTypesFilters` here and render it without storing it in a variable first